examples/DataApi-Water-And-Energy-Usage/template/modules/shared_data/src/main/scala/com/my/currency/shared_data/calculated_state/CalculatedStateService.scala

```diff
@@ -0,0 +1,71 @@
+package com.my.currency.shared_data.calculated_state
+
+import cats.effect.Ref
+import cats.effect.kernel.Async
+import cats.syntax.functor.toFunctorOps
+import com.my.currency.shared_data.types.Types.UsageUpdateCalculatedState
+import io.circe.Json
+import io.circe.syntax.EncoderOps
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.security.hash.Hash
+
+import java.nio.charset.StandardCharsets
+
+trait CalculatedStateService[F[_]] {
+  def getCalculatedState: F[CalculatedState]
+
+  def setCalculatedState(
+    snapshotOrdinal: SnapshotOrdinal,
+    state          : UsageUpdateCalculatedState
+  ): F[Boolean]
+
+  def hashCalculatedState(
+    state: UsageUpdateCalculatedState
+  ): F[Hash]
+}
+
+object CalculatedStateService {
+  def make[F[_] : Async]: F[CalculatedStateService[F]] = {
+    Ref.of[F, CalculatedState](CalculatedState.empty).map { stateRef =>
+      new CalculatedStateService[F] {
+        override def getCalculatedState: F[CalculatedState] = stateRef.get
+
+        override def setCalculatedState(
+          snapshotOrdinal: SnapshotOrdinal,
+          state          : UsageUpdateCalculatedState
+        ): F[Boolean] =
+          stateRef.update { currentState =>
+            val updatedDevices = state.devices.foldLeft(currentState.state.devices) {
+              case (acc, (address, value)) =>
+                acc.updated(address, value)
+            }
+
+            CalculatedState(
+              snapshotOrdinal,
+              UsageUpdateCalculatedState(updatedDevices)
+            )
+          }.as(true)
+
+        override def hashCalculatedState(
```

Comment:
This function feels like it doesn't belong here. See my suggestion for companion object for `UsageUpdateCalculatedState`

Reply:
Done